libraries/AP_RangeFinder/AP_RangeFinder_Benewake.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include "AP_RangeFinder_Benewake.h"
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <ctype.h>
+#include <AP_HAL/utility/sparse-endian.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define BENEWAKE_FRAME_HEADER 0x59
+#define BENEWAKE_FRAME_LENGTH 9
+
+// format of serial packets received from benewake lidar
+//
+// Data Bit             Definition      Description
+// ------------------------------------------------
+// byte 0               Frame header    0x59
+// byte 1               Frame header    0x59
+// byte 2               DIST_L          Distance (in cm) low 8 bits
+// byte 3               DIST_H          Distance (in cm) high 8 bits
+// byte 4               STRENGTH_L      Strength low 8 bits
+// byte 5               STRENGTH_H      Strength high 8 bits
+// byte 6 (TF02)        SIG             Reliability in 8 levels, 7 & 8 means reliable
+// byte 6 (TFmini)      Distance Mode   0x02 for short distance (mm), 0x07 for long distance (cm)
+// byte 7 (TF02 only)   TIME            Exposure time in two levels 0x03 and 0x06
```

Comment:
I think that it is better to know that it is 0 (without looking at the design sheet) for checksum check.
byte 7 (TFmini)                      0x00

Reply:
hmm.. this makes me think that we could actually auto detect if it's a TF02 or TFmini by looking at one of these fields..

Reply:
After a benchtest, it seems like byte 7 is always zero on both the tfmini and the tf02.. so not useful for automatically determining which lidar is being used.